packages/wrapped-keys/src/lib/api/batch-generate-private-keys.ts

```diff
@@ -0,0 +1,73 @@
+import { getKeyTypeFromNetwork } from './utils';
+import { batchGenerateKeysWithLitAction } from '../lit-actions-client';
+import { getLitActionCodeOrCidCommon } from '../lit-actions-client/utils';
+import { storePrivateKey } from '../service-client';
+import {
+  BatchGeneratePrivateKeysActionResult,
+  BatchGeneratePrivateKeysParams,
+  BatchGeneratePrivateKeysResult,
+} from '../types';
+import {
+  getFirstSessionSig,
+  getPkpAccessControlCondition,
+  getPkpAddressFromSessionSig,
+} from '../utils';
+
+/**
+ *  TODO: Document batch behaviour
+ * @param { BatchGeneratePrivateKeysParams } params Parameters to use for generating keys and optionally signing messages
+ *
+ * @returns { Promise<BatchGeneratePrivateKeysResult> } - The generated keys and, optionally, signed messages
+ */
+export async function batchGeneratePrivateKeys(
+  params: BatchGeneratePrivateKeysParams
+): Promise<BatchGeneratePrivateKeysResult> {
+  const { pkpSessionSigs, litNodeClient } = params;
+
+  const sessionSig = getFirstSessionSig(pkpSessionSigs);
+  const pkpAddress = getPkpAddressFromSessionSig(sessionSig);
+
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(pkpAddress);
+
+  const { litActionCode, litActionIpfsCid } = getLitActionCodeOrCidCommon(
+    'batchGenerateEncryptedKeys'
+  );
+
+  const actionResults = await batchGenerateKeysWithLitAction({
+    ...params,
+    litActionIpfsCid: litActionCode ? undefined : litActionIpfsCid,
+    litActionCode: litActionCode ? litActionCode : undefined,
+    accessControlConditions: [allowPkpAddressToDecrypt],
+    pkpSessionSigs,
+  });
+
+  return Promise.all(
```

Comment:
If any of these calls fails, the entire execution will exit even if the LA was successful and we only need to re-try storing the results using `storePrivateKey()`. A retry on network errors would be good to add at some point since the alternative is that the consumer re-runs the entire process from the top again, even though we might have valid signatures and ciphertexts in scope.

Reply:
Cool we plan to update this later with a retry and NOT now?

Reply:
:+1: yes, just making a note about the failure mode